native/src/main/java/io/ballerina/stdlib/etl/nativeimpl/EtlSecurity.java

```diff
@@ -0,0 +1,154 @@
+                    encryptedData.put(keyField, StringUtils.fromString(encryptedBase64));
+                } else {
+                    encryptedData.put(keyField, data.get(keyField));
+                }
+            }
+            encryptedDataset.append(encryptedData);
+        }
+        return encryptedDataset;
+    }
+
+    public static Object decryptData(BArray dataset, BArray fieldNames, BString key, BTypedesc returnType) {
+        BArray decryptedDataset = initializeBArray(returnType);
+        byte[] decryptKey = Base64.getDecoder().decode(key.getValue());
+
+        Cipher cipher;
+        try {
+            cipher = Cipher.getInstance("AES/ECB/PKCS5Padding");
+            cipher.init(Cipher.DECRYPT_MODE, new SecretKeySpec(decryptKey, "AES"));
+        } catch (Exception e) {
+            return ErrorUtils.createError(e.getMessage());
+        }
+        for (int i = 0; i < dataset.size(); i++) {
+            BMap<BString, Object> data = (BMap<BString, Object>) dataset.get(i);
+            BMap<BString, Object> decryptedData = initializeBMap(returnType);
+            for (int j = 0; j < fieldNames.size(); j++) {
+                BString fieldName = (BString) fieldNames.get(j);
+                if (!data.containsKey(fieldName)) {
+                    return ErrorUtils.createFieldNotFoundError(fieldName);
+                }
+            }
+            BString[] keys = data.getKeys();
+            for (BString keyField : keys) {
+                if (contains(fieldNames, keyField)) {
+                    String value = data.get(keyField).toString();
+                    byte[] decryptedBytes = null;
+                    try {
+                        decryptedBytes = cipher.doFinal(Base64.getDecoder().decode(value));
+                    } catch (IllegalBlockSizeException e) {
+                        return ErrorUtils.createError(e.getMessage());
+                    } catch (BadPaddingException e) {
+                        return ErrorUtils.createError(e.getMessage());
+                    }
+                    String decryptedValue = new String(decryptedBytes, StandardCharsets.UTF_8);
+                    decryptedData.put(keyField, StringUtils.fromString(decryptedValue));
+                } else {
+                    decryptedData.put(keyField, data.get(keyField));
+                }
+            }
+            decryptedDataset.append(decryptedData);
+        }
+        return decryptedDataset;
+    }
+
+    public static Object maskSensitiveData(Environment env, BArray dataset, BString maskCharacter, BString modelName,
+            BTypedesc returnType) {
+        Object[] args = new Object[] { dataset, maskCharacter, modelName, returnType };
+        Object clientResponse = env.getRuntime().callFunction(env.getCurrentModule(), "maskSensitiveDataFunc", null,
+                args);
+        if (TypeUtils.getType(clientResponse).getName().equals("ClientConnectorError")) {
```

Comment:
Instead of using `String` `"ClientConnectorError"` here, define a constant (`public static final`) variable and use it. Check similar places.

```suggestion
        if (TypeUtils.getType(clientResponse).getName().equals("ClientConnectorError")) {
```